### PR TITLE
Deprecate session key attribute in XmlSwitchPipe

### LINKED ISSUE #2095: Feature request: Simplify XmlSwitchPipe with session keys
**Describe the issue**
The XmlSwitchPipe has an attribute sessionKey and getInputFromSessionKey, which I find quite confusing, as they have different behaviors. I think usage would be simplified if the sessionKey-attribute could be deprecated in some way, using the getInputFromSessionKey. I'm not sure how this could be done while keeping compatbile with the current XmlSwitchPipe-implementation. 

**Reporter**
Ricardo

**To Reproduce**

**Screenshots**

**Configuration**
```
Please provide the configuration of the Pipe or Receiver with the problem.
```

**Input**
```
Please provide an example of the input message. 
Alternatively, provide a Ladybug report.
```

**Environment**
```
Please go to the console of your Frank, click on 'Information' and copy-paste the results here. 
```

**Additional Environment**
 - DBMS: [e.g. Oracle, MSSQL, MariaDB, MySql, PostgreSQL]
 - Browser: [e.g. Chrome, Safari, Edge, Firefox]

**Additional Context**
Add any other context about the problem here. (f.e. ladybug report / test adapter with larva test)

### PR DESCRIPTION
**Pull request by alisihab**
Closes #2095

### CODE REVIEW

File: core/src/main/java/nl/nn/adapterframework/pipes/XmlSwitch.java [hunk 1]
```diff
@@ -231,52 +218,36 @@ public void setServiceSelectionStylesheetFilename(String newServiceSelectionStyl
...
+	@ConfigurationWarning("Please use 'getInputFromSessionKey' attribute instead")
 	@IbisDoc({"4", "Name of the key in the <code>PipeLineSession</code> to retrieve the input message from, if a styleSheetName or a xpathExpression is specified. " + 
 					"If no styleSheetName or xpathExpression is specified, the value of the session variable is used as the name of the forward. " + 
 					"If none of sessionKey, styleSheetName or xpathExpression are specified, the element name of the root node of the input message is taken as the name of forward.", ""})
 	public void setSessionKey(String sessionKey){
 		this.sessionKey = sessionKey;
-	}
-	public String getSessionKey(){
-		return sessionKey;
+		this.forwardNameSessionKey = sessionKey;
```
Comment (gvanbrakel) [inline core/src/main/java/nl/nn/adapterframework/pipes/XmlSwitch.java:234]: you cannot assign sessionKey to forwardNameSessionKey, i.m.o.

**Review by gvanbrakel — CHANGES_REQUESTED** [review]

File: core/src/main/java/nl/nn/adapterframework/pipes/XmlSwitch.java (hunk 1 above)
Comment (alisihab) [reply core/src/main/java/nl/nn/adapterframework/pipes/XmlSwitch.java:229]: Assignment is for backwards compability. 'sessionKey' attribute could be set by frank developers. Keeping the support for the sessionKey attribute requires either setting the forwardNameSessionKey or having additional if statement in doPipe.

Comment (alisihab) [reply core/src/main/java/nl/nn/adapterframework/pipes/XmlSwitch.java:229]: Well, you are right actually!

File: core/src/main/java/nl/nn/adapterframework/pipes/XmlSwitch.java [hunk 2]
```diff
@@ -85,37 +87,34 @@ public void configure() throws ConfigurationException {
 				ConfigurationWarnings.add(this, log, "has a emptyForwardName attribute. However, this forward ["+getEmptyForwardName()+"] is not configured.");
 			}
 		}
-
+		if(StringUtils.isNotEmpty(getGetInputFromSessionKey()) && StringUtils.isNotEmpty(getSessionKey())) {
+			throw new ConfigurationException("cannot have both getInputFromSessionKey and a sessionKey specified");
```
Comment (gvanbrakel) [inline core/src/main/java/nl/nn/adapterframework/pipes/XmlSwitch.java:91]: Why not?

File: core/src/main/java/nl/nn/adapterframework/pipes/XmlSwitch.java [hunk 3]
```diff
@@ -152,40 +151,38 @@ public void stop() {
...
 			} catch (Throwable e) {
 				throw new PipeRunException(this, getLogPrefix(session) + "got exception on transformation", e);
 			}
-		} else {
+		} else if(StringUtils.isNotEmpty(getSessionKey())) {
 			try {
-				if (StringUtils.isNotEmpty(getSessionKey())) {
-					forward = session.getMessage(sessionKey).asString();
-				} else {
-					forward = message.asString();
```
Comment (gvanbrakel) [inline core/src/main/java/nl/nn/adapterframework/pipes/XmlSwitch.java:179]: what happened to this branch?

**Review by gvanbrakel — CHANGES_REQUESTED** [review]

File: core/src/main/java/nl/nn/adapterframework/pipes/XmlSwitch.java (hunk 3 above)
Comment (alisihab) [reply core/src/main/java/nl/nn/adapterframework/pipes/XmlSwitch.java:179]: discussed

File: core/src/main/java/nl/nn/adapterframework/pipes/XmlSwitch.java [hunk 4]
```diff
@@ -152,40 +148,37 @@ public void stop() {
...
+				forward = session.getMessage(getForwardNameSessionKey()).asString();
+			} catch (IOException e) {
+				throw new PipeRunException(this, getLogPrefix(session)+"cannot open stream", e);
+			}
+		} else if(!(StringUtils.isEmpty(getXpathExpression()) && StringUtils.isEmpty(getStyleSheetName())) || StringUtils.isEmpty(getSessionKey())) {
 			try {
 				Map<String,Object> parametervalues = null;
-				parameterList =  getParameterList();
-				if (parameterList!=null) {
-					message.preserve();
```
Comment (nielsm5) [inline core/src/main/java/nl/nn/adapterframework/pipes/XmlSwitch.java:162]: Can the preserve be removed?

**Review by nielsm5 — CHANGES_REQUESTED** [review]

File: core/src/main/java/nl/nn/adapterframework/pipes/XmlSwitch.java (hunk 4 above)
Comment (alisihab) [reply core/src/main/java/nl/nn/adapterframework/pipes/XmlSwitch.java:162]: In this case, it cannot so put it back :)

**Review by nielsm5 — APPROVED** [review]

**Review by gvanbrakel — APPROVED** [review]

**Pull request merged**